sql/hive/src/test/scala/org/apache/spark/sql/hive/client/FiltersSuite.scala

```diff
@@ -108,6 +108,21 @@ class FiltersSuite extends SparkFunSuite with Logging with PlanTest {
     (a("datecol", DateType) =!= Literal(Date.valueOf("2019-01-01"))) :: Nil,
     "datecol != 2019-01-01")
 
+  filterTest("string filter with date type",
+    (a("strcol", StringType).cast(DateType) > Literal(Date.valueOf("2019-01-01"))) :: Nil,
+    "strcol > 2019-01-01")
```

Comment:
is it really safe? Seems it assumes hive will also cast the string col to date and compare with the date literal.